Project_1_EKF/src/FusionEKF.cpp

```diff
@@ -48,68 +51,82 @@ FusionEKF::~FusionEKF() {}
 void FusionEKF::ProcessMeasurement(const MeasurementPackage &measurement_pack) {
 
 
-  /*****************************************************************************
-   *  Initialization
-   ****************************************************************************/
-  if (!is_initialized_) {
+    /*****************************************************************************
+     *  Initialization
+     ****************************************************************************/
+    if (!is_initialized_) {
+        /**
+        TODO:
+          * Initialize the state ekf_.x_ with the first measurement.
+          * Create the covariance matrix.
+          * Remember: you'll need to convert radar from polar to cartesian coordinates.
+        */
+        // first measurement
+        cout << "EKF: " << endl;
+        ekf_.x_ = VectorXd(4);
+        ekf_.x_ << 1, 1, 1, 1;
+
+        if (measurement_pack.sensor_type_ == MeasurementPackage::RADAR) {
+            /**
+            Convert radar from polar to cartesian coordinates and initialize state.
+            */
+        } else if (measurement_pack.sensor_type_ == MeasurementPackage::LASER) {
+            /**
+            Initialize state.
+            */
+        }
+
+        // done initializing, no need to predict or update
+        is_initialized_ = true;
+        return;
+    }
+
+    /*****************************************************************************
+     *  Prediction
+     ****************************************************************************/
+    double dt = (measurement_pack.timestamp_ - previous_timestamp_) / 1000000.0; // micros to s
+    previous_timestamp_ = measurement_pack.timestamp_;
+
+    ekf_.F_ = getF(dt);
+    ekf_.Q_ = getQ(dt);
+    ekf_.Predict();
+
+    /*****************************************************************************
+     *  Update
+     ****************************************************************************/
+
     /**
-    TODO:
-      * Initialize the state ekf_.x_ with the first measurement.
```

Comment:
we could remove the word TODO, but keep the rest explanation as partial documentations, which helps reading in the future

Reply:
Yeah. Will do in the future. But this one is not done anyways:-)_

Reply:
Great. lol